host_core/native/hostcore_wasmcloud_native/Cargo.toml

```diff
@@ -10,8 +10,7 @@ path = "src/lib.rs"
 crate-type = ["dylib"]
 
 [dependencies]
-# Patched temporarily to fix TinyGo modules, branch will be removed in v0.63.0
-wasmcloud = { git = "https://github.com/wasmcloud/wasmcloud", branch = "fix/0.62-tinygo"}
+wasmcloud = { git = "https://github.com/wasmcloud/wasmcloud" }
```

Comment:
should this be pinned to a hash? If not, it could potentially pull a breaking change and break a build

Reply:
Since `Cargo.lock` is checked in, this will not happen unless explicitly requested via an update command. See https://github.com/rust-lang/cargo/blob/master/src/doc/src/reference/specifying-dependencies.md#specifying-dependencies-from-git-repositories

So if a breaking change is introduced to `main` of https://github.com/wasmCloud/wasmcloud, after `cargo update` is executed the test should fail. I would consider that a feature and not a bug, since that simplifies maintenance, meaning that we don't need to manually copy-paste revs every time we want to update this dep.

Eventually (once https://github.com/bytecodealliance/preview2-prototyping is released to crates.io), we will be able to simply consume releases here. I believe ETA on this is end of spring

Reply:
As long as the `Cargo.lock` won't auto update, I'm cool with it. I was wondering the same, like how cargo will pull in patch releases automatically